### PR TITLE
quiz results per user + percentage fix

### PR DESCRIPTION
**Pull request by Adam3004**

### CODE REVIEW

File: src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java [hunk 1]
```diff
@@ -43,51 +45,56 @@ public Optional<Quiz> createQuiz(String title, String description, UUID workspac
                 title,
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java:47]: Nitpick: 
I don't get it why this method return optional
Btw it seems like there are definitely too many Optionals in this class

File: src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java [hunk 2]
```diff
@@ -43,29 +45,43 @@ public Note asNote(NoteEntity entity) {
         );
     }
 
-    public QuizSimpleResult asBestSimpleResult(QuizEntity entity) {
-        if (entity.getBestNumberOfIncorrect() == null || entity.getBestNumberOfCorrect() == null) {
+    public QuizSimpleResult asBestSimpleResult(QuizEntity entity, String userId) {
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java:47]: Nitpick: 
I think this shouldn't be here as this class is just for mappings between domain nad persistance objects (but as long as we don't care too much about code quality you don't have to waste time for that - saying just as information for future changes)

File: src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java [hunk 3]
```diff
@@ -43,29 +45,43 @@ public Note asNote(NoteEntity entity) {
...
+        Optional<QuizResultsEntity> foundResults = entity.getQuizResults().stream()
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java]: Maybe something like:
```
return entity.getQuizResults().stream()
.filter(results -> results.getUserId().equals(userId))
.findFirst()
.map(this::asQuizSimpleBestResult)
.orElse(null);
```
WDYT?

File: src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java [hunk 4]
```diff
@@ -43,29 +45,43 @@ public Note asNote(NoteEntity entity) {
...
+        if (quizResults.getBestNumberOfIncorrect() == null || quizResults.getBestNumberOfCorrect() == null) {
+            return null;
+        }
+        return new QuizSimpleResult(quizResults.getBestNumberOfIncorrect(), quizResults.getBestNumberOfCorrect());
     }
 
-    public QuizSimpleResult asLastSimpleResult(QuizEntity entity) {
-        if (entity.getLastNumberOfIncorrect() == null || entity.getLastNumberOfCorrect() == null) {
+    public QuizSimpleResult asLastSimpleResult(QuizEntity entity, String userId) {
+        Optional<QuizResultsEntity> foundResults = entity.getQuizResults().stream()
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java]: Same as above?

File: src/main/java/com/brightpath/learnify/persistance/quiz/QuizEntity.java [hunk 5]
```diff
@@ -2,22 +2,25 @@
...
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
 import java.time.OffsetDateTime;
+import java.util.Set;
 import java.util.UUID;
 
 @Entity
-@Table(name = "quizes")
+@Table(name = "quizzes")
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/persistance/quiz/QuizEntity.java:23]: Good catch

**Review by krisx303 — APPROVED** [review]
LGTM, just little nitpicking from my side in the comments

File: src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java (hunk 3 above)
Comment (Adam3004) [reply src/main/java/com/brightpath/learnify/persistance/common/PersistentMapper.java]: nice idea

**Pull request merged**